Add additional information about the style guide in CONTRIBUTING.md. (#2078)
fsprojects/fantomas: CONTRIBUTING.md

However, this can only happen if you discuss your proposed changes early and often.
It's often better to check *before* contributing that you're setting off on the right path.

## Fixing style guide inconsistencies

Fantomas tries to keep up with the style guides, but as these are living documents, it can occur that something is listed in the style that Fantomas is not respecting.
In this case, please create an issue using our [online tool](https://fsprojects.github.io/fantomas-tools/#/).  
Copy the code snippet from the guide and add a link to the section of the guide that is not being respected.  

> The maintainers will then add the `bug (stylistic)` to confirm the bug may be fixed in Fantomas. In most cases, it may seem obvious that the case can be fixed.

Reviewer: ```suggestion
The maintainers will then add the `bug (stylistic)` to confirm the bug is fixable in Fantomas. In most cases, it may seem obvious that the case can be fixed.
```

("may be fixed" could mean "might already be fixed")

> However, in the past there have been changes to the style guide that Fantomas could not implement due to technical reason. Any guidance made beyond the boundaries of the untyped syntax tree cannot be guaranteed.

Reviewer: ```suggestion
However, in the past there have been changes to the style guide that Fantomas could not implement for technical reasons: Fantomas can only implement rules based on information entirely contained within the untyped syntax tree.
```

Original was unambiguous but had slightly odd grammar.

> ### Target the next minor of major branch

Reviewer: ```suggestion
### Target the next minor or major branch
```

When fixing a stylistic issue, please ask the maintainers what branch should be targeted. The rule of thumb is that the `master` branch is used for fixing `bug (soundness)` and will be used for revision releases.

> End users should be able to update to the latest revision of their current minor or major without fear.

Reviewer: ```suggestion
Strive to ensure that end users can always update to the latest patch revision of their current minor or major without fear.
```

Just added the official semver word "patch", and reworded the start away from "should" (which is a little ambiguous between "we want this to be true" or "we think this is true").

> If the style changed, user should only need to deal with this when explicitly [upgrading](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#updating-to-a-new-fantomas-version) to a new minor or major version.

Reviewer: ```suggestion
A user should only need to deal with style changes when they have explicitly [chosen to upgrade](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#updating-to-a-new-fantomas-version) to a new minor or major version.
```
